### PR TITLE
Refactored UInt and Derived Classes

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
Refactored UInt and Derived classes to use `class`, and in addition 

### Context of Change
Old code was outdated. Reduced dependencies and updated code. 

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After
Removed dependency for `BN.js` in favor of `BigInt`
Migrated `makeClass()` to `class`

## Test Plan
No new tests added, still passes all test

### CODE REVIEW

File: src/types/uint.ts [hunk 1]
```diff
@@ -1,64 +1,61 @@
...
+      : val.toString();
+  }
+
+  /**
+   * Get the value of the UInt represented by this.bytes
+   * 
+   * @returns the value
+   */
+  valueOf(): number | bigint {
+    throw new Error("Cannot get value of the UInt Base Class")
```
Comment (keefertaylor) [inline src/types/uint.ts:57]: Let's mark this class as abstract and make this method abstract? 
https://www.staging-typescript.org/docs/handbook/classes.html#abstract-classes

File: src/types/uint-16.ts [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
-    inherits: UInt,
-    statics: { width: 2 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 16 bit UInt
+ */
+class UInt16 extends UInt {
+  static readonly width: number = 2
```
Comment (keefertaylor) [inline src/types/uint-16.ts:7]: optional: I think this is pretty obvious but *maybe* a bit more obvious to write something like:
```
static readonly width: number = 16 / 8 // 2
```

Comment (keefertaylor) [inline src/types/uint-16.ts:7]: Can these members be marked private?

File: src/types/uint-32.ts [hunk 3]
```diff
@@ -1,12 +1,39 @@
...
-    inherits: UInt,
-    statics: { width: 4 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 32 bit UInt
+ */
+class UInt32 extends UInt {
+  static readonly width: number = 4
```
Comment (keefertaylor) [inline src/types/uint-32.ts:7]: same comment on private members

File: src/types/uint-64.ts [hunk 4]
```diff
@@ -1,50 +1,76 @@
...
-          if (!HEX_REGEX.test(arg)) {
-            throw new Error(`${arg} is not a valid UInt64 hex string`);
-          }
-        }
-        this._bytes = parseBytes(arg, Uint8Array);
+/**
+ * Derived UInt class for serializing/deserializing 64 bit UInt
+ */
+class UInt64 extends UInt {
+  static readonly width: number = 8;
```
Comment (keefertaylor) [inline src/types/uint-64.ts:9]: ditto

File: src/types/uint-16.ts (hunk 2 above)
Comment (natenichols) [reply src/types/uint-16.ts:7]: No, but can be marked `protected`

File: src/types/uint-32.ts (hunk 3 above)
Comment (natenichols) [reply src/types/uint-32.ts:7]: same

File: src/types/uint-64.ts (hunk 4 above)
Comment (natenichols) [reply src/types/uint-64.ts:9]: same

File: src/types/uint-16.ts [hunk 5]
```diff
@@ -1,12 +1,44 @@
...
-    inherits: UInt,
-    statics: { width: 2 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 16 bit UInt
+ */
+class UInt16 extends UInt {
+  protected static readonly width: number = 16 / 8 //2
```
Comment (keefertaylor) [inline src/types/uint-16.ts:8]:
```suggestion
  protected static readonly width: number = 16 / 8 // 2 bytes
```

File: src/types/uint-32.ts [hunk 6]
```diff
@@ -1,12 +1,44 @@
...
-    inherits: UInt,
-    statics: { width: 4 },
-  },
-  undefined
-);
+/**
+ * Derived UInt class for serializing/deserializing 32 bit UInt
+ */
+class UInt32 extends UInt {
+  protected static readonly width: number = 32 / 8 //4
```
Comment (keefertaylor) [inline src/types/uint-32.ts:8]:
```suggestion
  protected static readonly width: number = 32 / 8 // 4 bytes
```

File: src/types/uint-32.ts [hunk 7]
```diff
@@ -1,12 +1,44 @@
...
+  static fromParser(parser: BinaryParser): UInt {
+    return new UInt32(parser.read(UInt32.width));
+  }
+
+  /**
+   * Construct a UInt32 object from a number
+   * 
+   * @param val UInt32 object or number
+   */
+  static from(val: UInt32 | number): UInt32 {
```
Comment (keefertaylor) [inline src/types/uint-32.ts:24]: Similar to other comments, I would type these out as `value` and `buffer` for clarity / readability. 

Some thoughts: https://softwareengineering.stackexchange.com/questions/72593/how-to-abbreviate-variable-names#:~:text=The%20standard%20I%20use%20is,lessens%20their%20ability%20to%20communicate.

File: src/types/uint-8.ts [hunk 8]
```diff
@@ -1,12 +1,44 @@
...
+      return val;
+    }
+
+    let buf = Buffer.alloc(UInt8.width);
+    buf.writeUInt8(val);
+    return new UInt8(buf);
+  }
+
+  /**
+   * get the value of a UInt8 object
```
Comment (keefertaylor) [inline src/types/uint-8.ts:35]: nit
```suggestion
   * Get the value of a UInt8 object
```

**Pull request merged**